core/mutator/mutator.go

```diff
@@ -47,9 +47,9 @@ var (
 
 // Mutator verifies mutations and transforms values in the map.
 type Mutator interface {
-	// CheckMutation verifies that this is a valid mutation for this item and
+	// Mutate verifies that this is a valid mutation for this item and
 	// applies mutation to value.
-	Mutate(value, mutation []byte) ([]byte, error)
```

Comment:
To make this more generic, should we consider making the mutation a generic protobuf?